src/metabase/driver/common/parameters/dates.clj

```diff
@@ -110,43 +138,48 @@
 (def ^:private relative-date-string-decoders
   [{:parser #(= % "today")
     :range  (fn [_ dt]
-              {:start dt,
-               :end   dt})
+              (let [dt-res (.toLocalDate ^LocalDateTime dt)]
+                {:start dt-res,
+                 :end   dt-res}))
     :filter (fn [_ field] [:= [:datetime-field field :day] [:relative-datetime :current]])}
 
    {:parser #(= % "yesterday")
     :range  (fn [_ dt]
-              {:start (t/minus dt (t/days 1))
-               :end   (t/minus dt (t/days 1))})
+              (let [dt-res (.toLocalDate ^LocalDateTime dt)]
+                {:start (t/minus dt-res (t/days 1))
+                 :end   (t/minus dt-res (t/days 1))}))
     :filter (fn [_ field] [:= [:datetime-field field :day] [:relative-datetime -1 :day]])}
 
    ;; adding a tilde (~) at the end of a past<n><unit> filter means we should include the current day/etc.
    ;; e.g. past30days  = past 30 days, not including partial data for today ({:include-current false})
    ;;      past30days~ = past 30 days, *including* partial data for today   ({:include-current true})
-   {:parser (regex->parser #"past([0-9]+)(day|week|month|year)s(~?)", [:int-value :unit :include-current?])
+   {:parser (regex->parser #"past([0-9]+)(second|minute|hour|day|week|month|year)s(~?)", [:int-value :unit :include-current?])
     :range  (fn [{:keys [unit int-value unit-range to-period include-current?]} dt]
-              (unit-range (t/minus dt (to-period int-value))
-                          (t/minus dt (to-period (if (seq include-current?) 0 1)))))
+              (let [dt-res (maybe-reduce-resolution unit dt)]
+                (unit-range (t/minus dt-res (to-period int-value))
+                            (t/minus dt-res (to-period (if (seq include-current?) 0 1))))))
     :filter (fn [{:keys [unit int-value include-current?]} field]
               [:time-interval field (- int-value) (keyword unit) {:include-current (boolean (seq include-current?))}])}
 
-   {:parser (regex->parser #"next([0-9]+)(day|week|month|year)s(~?)" [:int-value :unit :include-current?])
+   {:parser (regex->parser #"next([0-9]+)(second|minute|hour|day|week|month|year)s(~?)" [:int-value :unit :include-current?])
     :range  (fn [{:keys [unit int-value unit-range to-period include-current?]} dt]
-              (unit-range (t/plus dt (to-period (if (seq include-current?) 0 1)))
-                          (t/plus dt (to-period int-value))))
+              (let [dt-res (maybe-reduce-resolution unit dt)]
+                (unit-range (t/plus dt-res (to-period (if (seq include-current?) 0 1)))
+                            (t/plus dt-res (to-period int-value)))))
     :filter (fn [{:keys [unit int-value]} field]
               [:time-interval field int-value (keyword unit)])}
 
    {:parser (regex->parser #"last(day|week|month|year)" [:unit])
```

Comment:
what can you not do `lasthour`/`lastminute`/`lastsecond`? It seems like if you're adding support for `next` you should add it for `last` too

Reply:
`this` too. Why can't I do `thishour`?

Reply:
Added to backend, but no way to do this on the frontend (see longer comment).